### DISCUSSION TITLE
Amazon echo devices

### DISCUSSION THREAD

**Discussion by sevenlayercookie** [philippe44/AirConnect#530] [Q&A]
Does this work for Amazon Echo devices? Dot, etc.

**Comment by philippe44** [comment]
No, these are not UPnP or chromecast capable.

**Comment by Clem-** [comment]
I am also looking to a way to sync with Amazon Echo devices using the unofficial Alexa API.
I am not familiar with C and was wondering if you have a piece of software that can be used to spawn virtual AirPlay receivers without the Chromcast/UPnP sync and discovery?

Thank you for your really great tools !

**Comment by philippe44** [comment]
You will only be able to sync real AirPlay receivers.

**Reply by Clem-** [reply]
> It will detect UPnP/Sonos/Chromecast players, create as many virtual AirPlay devices as needed, and act as a bridge.

What I am looking for is to use AirPlay from an Apple device to a virtual receiver, get the stream metadata (artist, title, volume) and then use the Alexa API to play that song or change the volume.
There would not be any music stream going on. Just a pseudo-synchronization based on metadata.

**Comment by philippe44** [comment]
You can use this https://github.com/philippe44/libraop, although the README only refers to it as an AirPlay server, there is an AirPlay client as well. I've reorganized all my code sometimes ago to make that modular and I use this library in all my applications. Now, you need to be confortable with C, unfortunately.

I'd recommend starting from aircast or airupnp and stripping out "slowly" the upnp/chromecast parts. It's reasonanbly easy to do as the two "domains" are very separated. There is a main thread that do the upnp/cc search and for each one discovered spins an AirPlay instance. You could start from airupnp/aircast.c and remove calls to upnp/cc function and compile it, step by step. At one point, you'll be able to remove the upnp/cc libraries themselves and you'll be close to your goal. 

Now, if you want to use it as an external library, ala Python or JS, this is not the right system.
